experiments/duke/graphql_endpoint/graphql/resolvers.go

```diff
@@ -16,52 +20,119 @@ func personResolver(params graphql.ResolveParams) (interface{}, error) {
 	return person, err
 }
 
-func peopleResolver(params graphql.ResolveParams) (interface{}, error) {
-	size := params.Args["size"].(int)
-	from := params.Args["from"].(int)
+// NOTE: this duplicates structure here:
+// var PersonFilter *graphql.InputObject
+// not sure best way to go about this
+type CommonFilter struct {
+	Limit  int
+	Offset int
+	Query  string
+}
+
+// NOTE: these aren't different now, but dealing with
+// facets would probably make them different
+type PersonFilterParam struct {
+	Filter CommonFilter
+}
+
+type PublicationFilterParam struct {
+	Filter CommonFilter
+}
+
+type GrantFilterParam struct {
+	Filter CommonFilter
+}
+
+func convertPeopleFilter(params graphql.ResolveParams) (PersonFilterParam, error) {
+	result := PersonFilterParam{}
+	err := ms.Decode(params.Args, &result)
+	return result, err
+}
 
-	personList, err := elastic.FindPeople(size, from)
+func convertPublicationFilter(params graphql.ResolveParams) (PublicationFilterParam, error) {
+	// default values?
+	result := PublicationFilterParam{}
+	err := ms.Decode(params.Args, &result)
```

Comment:
hi @nymbyl  - it looks like you were able to use mapstructure. I do see your point about casting all of the values.